dyno-core/src/main/java/com/netflix/dyno/connectionpool/Host.java

```diff
@@ -35,27 +36,57 @@
 		Up, Down;
 	}
 	
-	public Host(String name, int port) {
-		this(name, port, Status.Down);
+	public Host(String hostname, int port) {
+		this(hostname, null, port, Status.Down);
 	}
 	
-	public Host(String name, Status status) {
-		this.name = name;
+	public Host(String hostname, Status status) {
+		this.hostname = hostname;
```

Comment:
you have one superset constructor which takes all possible arguments. You should redirect all the other constructors to that one like you do on line 39.